tools/integration_tests/readonly/list_objects_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Provides integration tests for file operations with --o=ro flag set.
+package readonly_test
+
+import (
+	"log"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/setup"
+)
+
+func TestListObjectsInBucket(t *testing.T) {
+	// ** Directory structure **
+	// testBucket
+	// testBucket/Test       -- Dir
+	// testBucket/Test1.txt  -- File
+
+	obj, err := os.ReadDir(setup.MntDir())
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if len(obj) != NumberOfObjectsInTestBucket {
+		t.Errorf("The number of objects in the current directory doesn't match.")
+	}
+
+	if obj[0].Name() != "Test" && obj[0].IsDir() != true {
```

Comment:
For reviewer: Other properties were FileInfo -> which has property fileName and Type and FileMode which is fileType only, both are included in this two comparison that's why I didn't mention.